src/fabric/src/fabric2_fdb.erl

```diff
@@ -864,6 +868,31 @@ bump_metadata_version(Tx) ->
     erlfdb:set_versionstamped_value(Tx, ?METADATA_VERSION_KEY, <<0:112>>).
 
 
+check_metadata_version(#{} = Db) ->
+    #{
+        tx := Tx,
+        layer_prefix := LayerPrefix,
+        name := DbName,
+        md_version := Version
+    } = Db,
+
+    AlreadyChecked = get(?PDICT_CHECKED_MD_IS_CURRENT),
+    if AlreadyChecked == true -> {current, Db}; true ->
+        case erlfdb:wait(erlfdb:get_ss(Tx, ?METADATA_VERSION_KEY)) of
+            Version ->
+                put(?PDICT_CHECKED_MD_IS_CURRENT, true),
+                % We want to set a read conflict on the db version as we'd want
+                % to to conflict with any writes to this particular db
+                DbPrefix = erlfdb_tuple:pack({?DBS, DbName}, LayerPrefix),
+                DbVersionKey = erlfdb_tuple:pack({?DB_VERSION}, DbPrefix),
+                erlfdb:add_read_conflict_key(Tx, DbVersionKey),
```

Comment:
Why add DbVersionKey as read_conflict_key if METADATA_VERSION_KEY is not changed?
If METADATA_VERSION_KEY hasn't changed, DbVersionKey certainly hasn't changed.

Reply:
This is so that any operation that updates the metadataVersion key doesn't cause a conflict for every transaction on other databases.